captum/attr/_core/lime.py

```diff
@@ -602,6 +622,31 @@ def default_perturb_func(original_inp, **kwargs):
     return torch.bernoulli(probs).to(device=device).long()
 
 
+def construct_feature_mask(feature_mask, formatted_inputs):
+    if feature_mask is None:
+        feature_mask, num_interp_features = _construct_default_feature_mask(
+            formatted_inputs
+        )
+    else:
+        feature_mask = _format_input(feature_mask)
+        min_interp_features = int(
+            min(torch.min(single_inp).item() for single_inp in feature_mask)
+        )
+        if min_interp_features != 0:
```

Comment:
Yeah, this would still technically work, since it would essentially create a feature vector of 100, but the extra features should be irrelevant, so results should match. But this would definitely be inefficient, so users should be careful of not leaving out feature values, which matches documentation. We could consider doing an extra step to remap non-consecutive IDs to a consecutive range, but it seems unnecessary if used correctly. I agree this warning is also not essential, and we could throw an exception, but since the current behavior is this way and it helps to avoid an extra step of just adjusting by a constant, it's probably fine to just leave it the same way for now.